zephyr/lib/dma.c

```diff
@@ -107,6 +107,26 @@ SHARED_DATA struct dma dma[] = {
 	.z_dev		= DEVICE_DT_GET(DT_NODELABEL(hda_link_out)),
 },
 #endif
+#ifdef CONFIG_SOC_SERIES_MIMX9_A55
+{
+	.plat_data = {
+		.dir = DMA_DIR_MEM_TO_DEV | DMA_DIR_DEV_TO_MEM,
+		.devs = DMA_DEV_SAI,
+		.channels = 64, /* TODO: this shouldn't be hardcoded */
```

Comment:
a device tree property?

Reply:
no such property ATM (`dma-channels` can't be used)

Reply:
this can be added later in DT and updated here when Zephyr side is ready.

Reply:
```suggestion
		.channels = 64, /* TODO: add a new DT property in Zephyr */
```

Reply:
About this: we do already have a DT property for this (there's 2 channel count-related ones actually). The issue is that they don't necessarily tell you what the max number of channels* is (which I'm assuming should be the value we want here). Anyways, for now I'd prefer sticking with the generic comment until we fix this via making use of (one) the already existing two properties.


(*) or the number of contiguous channels

Reply:
It's not because you're not sure yet how to fix this TODO that you should be holding back the information that you already have. Comments are cheap and you already wrote them just now. Why not have this knowledge directly in the source code? Readers rarely go all the way back to the code review. You may or may not be the one who will fix this later and even if it's you, you will most likely have forgotten half of it in a few months as we  typically all do.

Don't get me wrong: I hate pointlessly verbose comments that just paraphrase the code - which should ideally be so clear and easy to read that it does not require any comment. But this situation here looks like the exact opposite.

My 2 cents.

Reply:
Fair point, tweaked the comment to make it a bit more useful while still keeping it short